src/admin/index.tsx

```diff
@@ -28,25 +29,27 @@ const Index = () => (
       }}
       >
         <ScrollInfoProvider>
-          <Router>
-            <ModalProvider
-              classPrefix="payload"
-              zIndex={50}
-            >
-              <AuthProvider>
-                <PreferencesProvider>
-                  <SearchParamsProvider>
-                    <LocaleProvider>
-                      <StepNavProvider>
-                        <Routes />
-                      </StepNavProvider>
-                    </LocaleProvider>
-                  </SearchParamsProvider>
-                  <ModalContainer />
-                </PreferencesProvider>
-              </AuthProvider>
-            </ModalProvider>
-          </Router>
+          <Provider>
+            <Router>
+              <ModalProvider
+                classPrefix="payload"
+                zIndex={50}
+              >
+                <AuthProvider>
+                  <PreferencesProvider>
+                    <SearchParamsProvider>
+                      <LocaleProvider>
+                        <StepNavProvider>
+                          <Routes />
```

Comment:
The `<Provider />` should be nested directly above `<Routes />` so as to allow the user's custom provider components to access auth, preferences, modals, locales, search params, and step nav. As it stands now, custom provider components would have no access to those items.  Also, should we name that component `CustomProvider` or similar? Provider is quite generic.